src/decoder/unfiltering_buffer.rs

```diff
@@ -111,16 +124,27 @@ impl UnfilteringBuffer {
         self.current_start = 0;
         self.filled = 0;
         self.available = 0;
```

Comment:
I wonder if (maybe even as a separate PR, or at least a separate commit) it would help to change how `UnfilteringBuffer` is initialized.  In particular, I wonder if things would be simplified if `UnfilteringBuffer` would **always** have room for 2 mutable rows (2 times `rowlen` bytes).  IMHO this may be beneficial in the following ways:

* This way `prev_row` **always** exists (initially filled with zeros) and we don’t need to worry about special-casing unfiltering of the first row.
* This way there is no need for shifting/shuffling things around in `fn unfilter_ahead_row` / `fn unfilter_curr_row_in_scratch_buffer`, because the "scratch buffer" is **always** / **already** present at the beginning of the buffer.

WDYT?